File: Frontend/src/app/models/invoice.ts
export interface Invoice {
  invoiceId: number;
  clientName: string;
  totalAmount: number;
  status: 'paid' | 'unpaid' | 'pending';
  createdDate: string;
  clientId: string;
}

export interface GenerateInvoiceRequest {
  clientName: string;
}

export interface MarkPaidRequest {
  invoiceId: number;
}

File: FreelancerFlow.API/Models/Invoice.cs
using FreelancerFlow.API.Enums;

namespace FreelancerFlow.API.Models
{
    public class Invoice
    {
        public int InvoiceId { get; set; } // Primary Key

        public int ClientId { get; set; } // Foreign Key

        public Client Client { get; set; } // Navigation

        public double TotalAmount { get; set; } // Total price as double

        public InvoiceStatus Status { get; set; } = InvoiceStatus.Unpaid; // Status as enum

        public DateTime CreatedAt { get; set; } // Date created
    }
}
